src/js/jsx/shared/Datalist.jsx

```diff
@@ -451,6 +475,15 @@ define(function (require, exports, module) {
             }
         },
 
+        removeAutofillSuggestion: function () {
+            if (this.props.useAutofill) {
+                this.setState({
+                    suggestTitle: ""
+                });
+                this.forceUpdate();
```

Comment:
This is a bad code smell. `shouldComponentUpdate` really should not have side effects. Why is that necessary? What are other ways that we could get rid of this extra call? If I'm understanding this, I think the proposed control flow is too messy to commit.